standardtracer/propagation.go

```diff
@@ -44,9 +45,11 @@ func (p *splitTextPropagator) InjectSpan(
 		fieldNameSampled: strconv.FormatBool(sc.raw.Sampled),
 	}
 	sc.Lock()
-	splitTextCarrier.TraceAttributes = make(map[string]string, len(sc.raw.Attributes))
-	for k, v := range sc.raw.Attributes {
-		splitTextCarrier.TraceAttributes[k] = v
+	if l := len(sc.raw.Attributes); l > 0 {
+		splitTextCarrier.TraceAttributes = make(map[string]string, l)
```

Comment:
I think we have to decide what the semantics should be. I want to avoid a mandatory allocation if not using attributes at all, so I want the `nil` map to be valid for the carrier, which means that memory must be allocated sometimes (or the attributes dropped, which we probably don't want). The sane thing seems to be:

* if `nil`, create one
* if not `nil`, loop through and delete everything, then use it.

Or is there an explicit use case for adding to a nonempty map? I seems like that would invite surprises.